consul/src/main/java/com/linecorp/armeria/internal/consul/ConsulClientBuilder.java

```diff
@@ -0,0 +1,151 @@
+        checkState(!isUriComponentSet, "consulUri can't comes with other addressing options");
+        requireNonNull(consulUri, "consulUri");
+        final String path = consulUri.getPath();
+        checkArgument(!Strings.isNullOrEmpty(path) && !"/".equals(path),
+                      "consulUri has to contain version path");
+        this.consulUri = consulUri;
+        return this;
+    }
+
+    /**
+     * Sets the specified Consul's API service URI.
+     * The URI should include the Consul API version at path, like: {@code /v1}
+     * @param consulUri the URI of Consul API service, default: {@code http://127.0.0.1:8500/v1}
+     */
+    public ConsulClientBuilder consulUri(String consulUri) {
+        return consulUri(URI.create(requireNonNull(consulUri, "consulUri")));
+    }
+
+    /**
+     * Sets the specified Consul's API service protocol scheme.
+     * @param consulProtocol the protocol scheme of Consul API service, default: {@link SessionProtocol#HTTP}
+     */
+    public ConsulClientBuilder consulProtocol(SessionProtocol consulProtocol) {
+        checkState(consulUri == null, "consulProtocol can't comes with consulUri");
+        requireNonNull(consulProtocol, "consulProtocol");
+        this.consulProtocol = consulProtocol;
+        isUriComponentSet = true;
+        return this;
+    }
+
+    /**
+     * Sets the specified Consul's API service host address.
+     * @param consulAddress the host address of Consul API service, default: {@code 127.0.0.1}
+     */
+    public ConsulClientBuilder consulAddress(String consulAddress) {
+        checkState(consulUri == null, "consulAddress can't comes with consulUri");
+        requireNonNull(consulAddress, "consulAddress");
+        checkArgument(!consulAddress.isEmpty(), "consulAddress can't be empty");
+        this.consulAddress = consulAddress;
+        isUriComponentSet = true;
+        return this;
+    }
+
+    /**
+     * Sets the specified Consul's HTTP service port.
+     * @param consulPort the port of Consul agent, default: {@code 8500}
+     */
+    public ConsulClientBuilder consulPort(int consulPort) {
+        checkState(consulUri == null, "consulPort can't comes with consulUri");
+        checkArgument(consulPort > 0, "consulPort can't be zero or negative");
+        this.consulPort = consulPort;
+        isUriComponentSet = true;
+        return this;
+    }
+
+    /**
+     * Sets the specified Consul's API version.
+     * @param consulApiVersion the version of Consul API service, default: {@code /v1}
+     */
+    public ConsulClientBuilder consulApiVersion(String consulApiVersion) {
+        checkState(consulUri == null, "consulApiVersion can't comes with consulUri");
+        requireNonNull(consulApiVersion, "consulApiVersion");
+        checkArgument(!consulApiVersion.isEmpty(), "consulApiVersion can't be empty");
+        checkArgument(consulApiVersion.charAt(0) != '/',
+                      "consulApiVersion can't starts with '/'");
```

Comment:
If the version should not start with `/`, how about replacing the default value (`/v1`) in documentation with `v1` for reducing confusion.

Reply:
Oops! Ok.

Reply:
How about checking using a regular expression? e.g. `^v[0-9][-._a-zA-Z0-9]*$`